src/modules/nuttx/iotjs_module_i2c-nuttx.c

```diff
@@ -25,105 +25,109 @@
 
 
 #define I2C_DEFAULT_FREQUENCY 400000
+#define I2C_DEFAULT_ADDRESS_LENGTH 7
 
 struct iotjs_i2c_platform_data_s {
-  int device;
+  int bus;
   struct i2c_master_s* i2c_master;
   struct i2c_config_s config;
 };
 
-void i2c_create_platform_data(void* device, iotjs_i2c_t* i2c,
-                              iotjs_i2c_platform_data_t** ppdata) {
-  iotjs_i2c_platform_data_t* pdata = IOTJS_ALLOC(iotjs_i2c_platform_data_t);
+void i2c_create_platform_data(iotjs_i2c_t* i2c) {
+  IOTJS_VALIDATED_STRUCT_METHOD(iotjs_i2c_t, i2c);
 
-  pdata->device = *(int*)device;
-  pdata->i2c_master = NULL;
-  *ppdata = pdata;
+  _this->platform_data = IOTJS_ALLOC(iotjs_i2c_platform_data_t);
+
+  _this->platform_data->i2c_master = NULL;
 }
 
 void i2c_destroy_platform_data(iotjs_i2c_platform_data_t* pdata) {
   IOTJS_RELEASE(pdata);
 }
 
-#define I2C_WORKER_INIT_TEMPLATE                                            \
-  iotjs_i2c_reqwrap_t* req_wrap = iotjs_i2c_reqwrap_from_request(work_req); \
-  iotjs_i2c_reqdata_t* req_data = iotjs_i2c_reqwrap_data(req_wrap);
-
-#define IOTJS_I2C_METHOD_HEADER(arg)              \
-  IOTJS_VALIDATED_STRUCT_METHOD(iotjs_i2c_t, arg) \
+jerry_value_t iotjs_i2c_set_platform_config(iotjs_i2c_t* i2c,
+                                            const jerry_value_t jconfig) {
+  IOTJS_VALIDATED_STRUCT_METHOD(iotjs_i2c_t, i2c);
   iotjs_i2c_platform_data_t* platform_data = _this->platform_data;
 
+  DJS_GET_REQUIRED_CONF_VALUE(jconfig, platform_data->bus,
+                              IOTJS_MAGIC_STRING_BUS, number);
 
-void I2cSetAddress(iotjs_i2c_t* i2c, uint8_t address) {
-  IOTJS_I2C_METHOD_HEADER(i2c);
-  platform_data->config.address = address;
-  platform_data->config.addrlen = 7;
+  return jerry_create_undefined();
 }
 
-void OpenWorker(uv_work_t* work_req) {
-  I2C_WORKER_INIT_TEMPLATE;
-  iotjs_i2c_t* i2c = iotjs_i2c_instance_from_reqwrap(req_wrap);
+#define IOTJS_I2C_METHOD_HEADER(arg)                               \
```

Comment:
@hs0225 We use `IOTJS_` or `iotjs_` when it is global name, this macro seems local in C file. Please use `I2C_METHOD_HEADER`.